pkg/certs/util.go

```diff
@@ -68,3 +71,58 @@ func CertificateBundleToString(bundle []*x509.Certificate) string {
 	}
 	return strings.Join(output, "\n")
 }
+
+func ValidatePrivateKeyFormat(pemKey []byte) []error {
+	if len(pemKey) == 0 {
+		return []error{fmt.Errorf("required private key is empty")}
+	}
+	if _, err := keyutil.ParsePrivateKeyPEM(pemKey); err != nil {
+		return []error{fmt.Errorf("failed to parse the private key, %w", err)}
+	}
+	return nil
+}
+
+func ValidateServerCertDate(pem []byte) []error {
```

Comment:
godoc. This is not obvious what it does. Maybe `ValidateServerCertValidity`?

Reply:
Added. I also inlined the `parse` function as this was the only place it was used.